src/core/qgis_sip.h

```diff
@@ -197,6 +197,17 @@
  */
 #define SIP_DOC_TEMPLATE
 
+/*
+ * Sip supports the final keyword since version 4.19.0, earlier than that
+ * we will have build issues because it tries to override final methods.
+ */
+#if SIP_VERSION < 0x041300
+#if defined FINAL
+#undef FINAL
+#endif
+#define FINAL override
```

Comment:
missing #else

Reply:
No else required, that's defined by qgis.h